File: backend/src/main/java/com/github/fjbaldon/attendex/backend/dto/ActiveEventResponse.java
package com.github.fjbaldon.attendex.backend.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ActiveEventResponse {
    private Long id;
    private String eventName;
    private List<TimeSlotResponse> timeSlots;
}

File: capture/app/src/main/java/com/github/fjbaldon/attendex/capture/core/data/remote/ActiveEventResponse.kt
package com.github.fjbaldon.attendex.capture.core.data.remote

import kotlinx.serialization.Serializable

@Serializable
data class ActiveEventResponse(
    val id: Long,
    val name: String,
    val startDate: String,
    val endDate: String,
    val sessions: List<SessionResponse>,
    // The backend sends 'roster' here now, but your app fetches it separately.
    // You can ignore it here or refactor the app to use it.
    // For now, ignore it to prevent parsing errors if your JSON config is strict.
    val roster: List<RosterItemResponse> = emptyList()
)
